packages/app-store/hubspotothercalendar/lib/CalendarService.ts

```diff
@@ -0,0 +1,212 @@
+import * as hubspot from "@hubspot/api-client";
+import { BatchInputPublicAssociation } from "@hubspot/api-client/lib/codegen/crm/associations";
+import { PublicObjectSearchRequest } from "@hubspot/api-client/lib/codegen/crm/contacts";
+import { BatchInputSimplePublicObjectInput } from "@hubspot/api-client/lib/codegen/crm/objects";
+import { SimplePublicObjectInput } from "@hubspot/api-client/lib/codegen/crm/objects/meetings";
+import { Credential } from "@prisma/client";
+
+import { getLocation, getAdditionalNotes } from "@calcom/lib/CalEventParser";
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+import prisma from "@calcom/prisma";
+import type {
+  AdditionInformation,
+  Calendar,
+  CalendarEvent,
+  ConferenceData,
+  EventBusyDate,
+  IntegrationCalendar,
+  NewCalendarEventType,
+} from "@calcom/types/Calendar";
+
+import type { HubspotToken } from "../api/callback";
+
+const hubspotClient = new hubspot.Client();
+
+const client_id = process.env.HUBSPOT_CLIENT_ID;
+const client_secret = process.env.HUBSPOT_CLIENT_SECRET;
+
+export default class HubspotOtherCalendarService implements Calendar {
+  private url = "";
+  private integrationName = "";
+  private auth: { getToken: () => Promise<any> };
+  private log: typeof logger;
+
+  constructor(credential: Credential) {
+    this.integrationName = "hubspot_other_calendar";
+
+    this.auth = this.hubspotAuth(credential);
+
+    this.log = logger.getChildLogger({ prefix: [`[[lib] ${this.integrationName}`] });
+  }
+
+  private hubspotContactSearch = async (event: CalendarEvent) => {
+    const publicObjectSearchRequest: PublicObjectSearchRequest = {
+      filterGroups: event.attendees.map((attendee) => ({
+        filters: [
+          {
+            value: attendee.email,
+            propertyName: "email",
+            operator: "EQ",
+          },
+        ],
+      })),
+      sorts: ["hs_object_id"],
+      properties: ["hs_object_id", "email"],
+      limit: 10,
+      after: 0,
+    };
+
+    return await hubspotClient.crm.contacts.searchApi
+      .doSearch(publicObjectSearchRequest)
+      .then((apiResponse) => apiResponse.results);
+  };
+
+  private getHubspotMeetingBody = (event: CalendarEvent): string => {
```

Comment:
I thought this helper could be in `CalEventParser` but as it's just scoped to HubSpot I left it here.

Reply:
Yeah we can move it later